### PR TITLE
[Speech2Text Doc] Fix docs

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Fixes Speech to Text model docs and add tests

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/model_doc/speech_to_text.mdx [hunk 1]
```diff
@@ -47,23 +47,15 @@ be installed as follows: `apt install libsndfile1-dev`
...
 >>> processor = Speech2TextProcessor.from_pretrained("facebook/s2t-small-librispeech-asr")
 
 
->>> def map_to_array(batch):
-...     speech, _ = sf.read(batch["file"])
-...     batch["speech"] = speech
-...     return batch
-
-
 >>> ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
```
Comment (ydshieh) [inline docs/source/en/model_doc/speech_to_text.mdx:55]: I saw we use `hf-internal-testing/librispeech_asr_demo` in `PT_SPEECH_BASE_MODEL_SAMPLE` and in other PT speech models in `doc.py`. Not sure if there is any difference between these 2 datasets, but maybe use `librispeech_asr_demo` in all places is good (if this makes sense)

File: docs/source/en/model_doc/speech_to_text.mdx [hunk 2]
```diff
@@ -47,23 +47,15 @@ be installed as follows: `apt install libsndfile1-dev`
...
->>> ds = ds.map(map_to_array)
 
->>> inputs = processor(ds["speech"][0], sampling_rate=16_000, return_tensors="pt")
->>> generated_ids = model.generate(input_ids=inputs["input_features"], attention_mask=inputs["attention_mask"])
+>>> inputs = processor(ds[0]["audio"]["array"], sampling_rate=ds[0]["audio"]["sampling_rate"], return_tensors="pt")
+>>> generated_ids = model.generate(inputs["input_features"], attention_mask=inputs["attention_mask"])
 
 >>> transcription = processor.batch_decode(generated_ids)
 ```
```
Comment (ydshieh) [inline docs/source/en/model_doc/speech_to_text.mdx:61]: Better to show the result ..?

File: docs/source/en/model_doc/speech_to_text.mdx [hunk 3]
```diff
@@ -80,24 +72,15 @@ be installed as follows: `apt install libsndfile1-dev`
...
->>> ds = ds.map(map_to_array)
 
->>> inputs = processor(ds["speech"][0], sampling_rate=16_000, return_tensors="pt")
+>>> inputs = processor(ds[0]["audio"]["array"], sampling_rate=ds[0]["audio"]["sampling_rate"], return_tensors="pt")
 >>> generated_ids = model.generate(
-...     input_ids=inputs["input_features"],
+...     inputs["input_features"],
 ...     attention_mask=inputs["attention_mask"],
 ...     forced_bos_token_id=processor.tokenizer.lang_code_to_id["fr"],
 ... )
```
Comment (ydshieh) [inline docs/source/en/model_doc/speech_to_text.mdx:86]: Better to show the result (`transcription`)?

**Review by ydshieh — APPROVED** [review]
Thanks! Left a few comments.

I didn't run it locally, since it looks like there is currently some dataset issue?

**Comment by patrickvonplaten** [comment]
Great review - thanks @ydshieh

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Pull request merged**